cmd/kpod/create.go

```diff
@@ -0,0 +1,899 @@
+	}
+
+	runtimeSpec, err := createConfigToOCISpec(createConfig)
+	if err != nil {
+		return err
+	}
+
+	imageName, err := createImage.GetFQName()
+	if err != nil {
+		return err
+	}
+	fmt.Println(imageName)
+	imageID, err := createImage.GetImageID()
+	if err != nil {
+		return err
+	}
+	ctr, err := runtime.NewContainer(runtimeSpec, libpod.WithRootFSFromImage(imageID, imageName, false))
+	if err != nil {
+		return err
+	}
+
+	if err := ctr.Create(); err != nil {
+		return err
+	}
+
+	if c.String("cid-file") != "" {
+		libpod.WriteFile(ctr.ID(), c.String("cid-file"))
+		return nil
+	}
+	fmt.Printf("%s\n", ctr.ID())
+
+	return nil
+}
+
+/* The following funcs should land in parse.go */
+//
+//
+func stringSlicetoUint32Slice(inputSlice []string) ([]uint32, error) {
+	var outputSlice []uint32
+	for _, v := range inputSlice {
+		u, err := strconv.ParseUint(v, 10, 32)
+		if err != nil {
+			return outputSlice, err
+		}
+		outputSlice = append(outputSlice, uint32(u))
+	}
+	return outputSlice, nil
+}
+
+// Parses CLI options related to container creation into a config which can be
+// parsed into an OCI runtime spec
+func parseCreateOpts(c *cli.Context, runtime *libpod.Runtime) (*createConfig, error) {
+	var command []string
+	var memoryLimit, memoryReservation, memorySwap, memoryKernel int64
+	var blkioWeight uint16
+	var env []string
+	var labelValues []string
+	var uid, gid uint32
+	sysctl := make(map[string]string)
+	labels := make(map[string]string)
+
+	image := c.Args()[0]
+
+	if len(c.Args()) < 1 {
+		return nil, errors.Errorf("you just provide an image name")
+	}
+	if len(c.Args()) > 1 {
+		command = c.Args()[1:]
+	}
+
+	// LABEL VARIABLES
+	// TODO where should labels be verified to be x=y format
+	labelValues, labelErr := readKVStrings(c.StringSlice("label-file"), c.StringSlice("label"))
+	if labelErr != nil {
+		return &createConfig{}, errors.Wrapf(labelErr, "unable to process labels from --label and label-file")
+	}
+	// Process KEY=VALUE stringslice in string map for WithLabels func
+	if len(labelValues) > 0 {
+		for _, i := range labelValues {
+			spliti := strings.Split(i, "=")
+			labels[spliti[0]] = spliti[1]
+		}
+	}
+
+	// ENVIRONMENT VARIABLES
+	// TODO where should env variables be verified to be x=y format
+	env, err := readKVStrings(c.StringSlice("env-file"), c.StringSlice("env"))
+	if err != nil {
+		return &createConfig{}, errors.Wrapf(err, "unable to process variables from --env and --env-file")
+	}
+	// Add default environment variables if nothing defined
+	if len(env) == 0 {
+		env = append(env, defaultEnvVariables...)
+	}
+
+	if len(c.StringSlice("sysctl")) > 0 {
+		for _, inputSysctl := range c.StringSlice("sysctl") {
+			values := strings.Split(inputSysctl, "=")
+			sysctl[values[0]] = values[1]
+		}
+	}
+
+	groupAdd, err := stringSlicetoUint32Slice(c.StringSlice("group-add"))
+	if err != nil {
+		return &createConfig{}, errors.Wrapf(err, "invalid value for groups provided")
+	}
+
+	if c.String("user") != "" {
```

Comment:
This sounds like it should be handled in libpod, after the image rootfs is created? I can make a WithUser() to handle mapping into the container's /etc/passwd and /etc/group

Reply:
SGTM